### PR TITLE
feat(components): stepper

### PR DESCRIPTION
**Pull request by mhuet**
Introducing a new component, the Stepper to build multi-page forms 

**Please check if the PR fulfills these requirements**

- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: src/components/Stepper/Stepper.style.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import styled from 'styled-components';
+import tokens from '../../tokens';
+
+export const Stepper = styled.ol`
+	width: 20rem;
+	display: flex;
+	list-style: none;
```
Comment (frassinier) [inline src/components/Stepper/Stepper.style.ts:7]: You can use [role='list'] on ol which will automatically disable list style

```suggestion
```

File: src/components/Stepper/Stepper.style.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	align-items: flex-end;
+	flex-direction: column;
+
+	> li:first-of-type ~ div.line:not(:last-child) {
```
Comment (frassinier) [inline src/components/Stepper/Stepper.style.ts:11]:
```suggestion
	> :first-of-type ~ .line:not(:last-child) {
```

File: src/components/Stepper/Stepper.style.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+		height: 5rem;
+		width: 0.2rem;
+		margin-right: 0.9rem;
+		background-image: url("data:image/svg+xml,%3csvg width='100%25' height='100%25' xmlns='http://www.w3.org/2000/svg'%3e%3crect width='100%25' height='100%25' fill='none' stroke='%23d2d2d2' stroke-width='4' stroke-dasharray='0%25%2c40%25' stroke-dashoffset='0' stroke-linecap='square'/%3e%3c/svg%3e");
+	}
+
+	> li:first-of-type ~ :not(.in-progress ~ *):not(.in-progress):not(li) {
```
Comment (frassinier) [inline src/components/Stepper/Stepper.style.ts:18]: Could be nice to replace the last li by a class 
```suggestion
	> :first-of-type ~ :not(.in-progress ~ *):not(.in-progress):not(li) {
```

File: src/components/Stepper/Step/Step.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+};
+
+/**
+ Steps are the main elements for the stepper.
+ * */
+
+const Step: React.FC<StepProps> = ({ icon, index, title, className = '', ...rest }: StepProps) => {
+	return (
+		<S.Step role="status" aria-live="polite" {...rest} className={`step ${className || ''}`}>
+			<span className="step__index">{index}</span>
```
Comment (frassinier) [inline src/components/Stepper/Step/Step.tsx:25]: The step index doesn't need to appear in HTML, it's some CSS decoration.


```suggestion
```

It's more a pseudo-element
```css
.step[data-index]::before { content: attr(data-index) ". "; }
```

File: src/components/Stepper/Stepper.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import React, { PropsWithChildren } from 'react';
+import * as S from './Stepper.style';
+
+export type StepperProps = PropsWithChildren<any>;
+
+const Stepper: React.FC<StepperProps> = ({ children, ...rest }: StepperProps) => (
+	<S.Stepper>
+		{React.Children.map(children, (child, index) => [
```
Comment (frassinier) [inline src/components/Stepper/Stepper.tsx:8]: Check if you have children before

File: src/components/Stepper/Stepper.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+			<div className="line" />,
```
Comment (frassinier) [inline src/components/Stepper/Stepper.tsx:9]: Move this on top of the loop

```suggestion
			<div className="stepper__progress-bar" />,
```

File: src/components/Stepper/Stepper.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+			React.cloneElement(child, { index: index + 1, ...rest }),
```
Comment (frassinier) [inline src/components/Stepper/Stepper.tsx:10]: You don't need to spread Stepper props here, spread into `<S.Stepper {...rest}>`

**Review by frassinier — COMMENTED** [review]
Excellent PR 💯

File: src/components/Stepper/Step/variations/Step.error.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
```
[the remark this answers is not in the document]
Comment (ncomont) [reply src/components/Stepper/Step/variations/Step.error.tsx:1]: image
wow, dumb copy paste from other components, sorry

File: src/components/Stepper/Step/variations/Step.validated.tsx [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+import Step from '../Step';
+import tokens from '../../../../tokens';
+
+const size = 2;
+
+const StepValidated = styled(Step).attrs({
+	icon: 'talend-check-circle',
+})`
+	.step__icon,
+	.step__icon > svg {
```
[the remark this answers is not in the document]
Comment (ncomont) [reply src/components/Stepper/Step/variations/Step.validated.tsx:12]: you mean ? .tc-svg-icon

File: src/components/Stepper/Stepper.tsx [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import React, { PropsWithChildren } from 'react';
+import * as S from './Stepper.style';
+
+export type StepperProps = PropsWithChildren<any>;
+
+const Stepper: React.FC<StepperProps> = ({ children, ...rest }: StepperProps) => (
+	<S.Stepper {...rest}>
+		{children &&
+			React.Children.map(children, (child, index) => [
+				<div className="stepper__progress-bar" />,
```
Comment (frassinier) [inline src/components/Stepper/Stepper.tsx:10]: Does it mean you render a `.stepper__progress-bar` next to each step?

File: src/components/Stepper/docs/Stepper.overview.stories.mdx [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+_(coming later)_
+
+## Usage
+
+_Add demo_
+
+## Accessibility
+
+- Press <kbd>TABS</kbd> to focus is on each steps (icon + label) meaning max focus will be 15rem
+- Press <kbd>SHIFT+TABS</kbd> to come back to the previous step
```
Comment (frassinier) [inline src/components/Stepper/docs/Stepper.overview.stories.mdx:103]:
```suggestion
- Press <kbd>Shift</kbd>+<kbd>Tabs</kbd> to come back to the previous step
```

File: src/components/Stepper/docs/Stepper.step.stories.mdx [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+	alt="hovering on"
+/>
+
+## Usage
+
+_To do_
+
+## Accessibility
+
+- Press <kbd>TABS</kbd> to focus is on each steps (icon + label) meaning max focus will be 15rem
```
Comment (frassinier) [inline src/components/Stepper/docs/Stepper.step.stories.mdx:127]:
```suggestion
- Press <kbd>Tabs</kbd> to focus is on each steps (icon + label) meaning max focus will be 15rem
```

File: src/components/Stepper/Stepper.tsx [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+				React.cloneElement(child, { 'data-index': index + 1 }),
```
Comment (frassinier) [inline src/components/Stepper/Stepper.tsx:11]: `<li>` should be here
```suggestion
				(<li>{React.cloneElement(child, { 'data-index': index + 1 })}</li>),
```

File: src/components/Stepper/docs/Stepper.overview.stories.mdx [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+**Animation**
+
+_(coming later)_
+
+## Usage
+
+_Add demo_
+
+## Accessibility
+
```
Comment (frassinier) [inline src/components/Stepper/docs/Stepper.overview.stories.mdx:101]:
```suggestion

Stepper should follow related [WAI guidelines](https://www.w3.org/WAI/tutorials/forms/multi-page/)
```

File: src/components/Skeleton/index.ts [hunk 15]
```diff
@@ -3,11 +3,11 @@ import SkeletonButton from './variations/Skeleton.Button';
 import SkeletonIconButton from './variations/Skeleton.IconButton';
 
 const SkeletonComponent = Skeleton as typeof Skeleton & {
-	Button: typeof SkeletonButton;
 	Icon: typeof SkeletonIconButton;
+	Button: typeof SkeletonButton;
```
[the remark this answers is not in the document]
Comment (ncomont) [reply src/components/Skeleton/index.ts:7]: i pushed some code in order to add the skeleton step, then i removed it, but this change has not be reverted

File: src/components/Stepper/Stepper.tsx (hunk 10 above)
Comment (ncomont) [reply src/components/Stepper/Stepper.tsx:10]: yep

**Review by frassinier — APPROVED** [review]

**Pull request merged**